overlay_manager.js

```diff
@@ -0,0 +1,222 @@
+        
+        let that = this;
+        function set_val() {
+            if (popup_dialogue._map && popup_dialogue._symbol) {
+                let val = popup_dialogue._input.value;
+                if (val && !isNaN(val) && Number.isInteger(+val) && val > 0) {
+                    popup_dialogue._map[popup_dialogue._symbol] = val;
+                    popup_dialogue._hide();
+                    that.prompt_define_symbol();
+                    return;
+                }
+            }
+            popup_dialogue._input.setCustomValidity('Invalid, not an integer');
+        }
+        this.popup_dialogue._input.addEventListener('keypress', (ev) => {
+            if (ev.which === 13)
+                set_val();
+        });
+
+        let footer_bar = createElement('div', '', ['modal_footer_bar'],
+            this.popup_dialogue);
+        let confirm_button = createElement('div', '',
+            ['button', 'modal_confirm_button'], footer_bar);
+        confirm_button.addEventListener('click', (ev) => { set_val(); });
+        let confirm_button_text = createElement('span', '', [], confirm_button);
+        confirm_button_text.innerText = 'Confirm';
+        createElement('div', '', ['clearfix'], footer_bar);
+
+        this.popup_dialogue._show = function (symbol, map) {
+            this.style.display = 'block';
+            popup_dialogue._title.innerText = 'Define symbol ' + symbol;
+            popup_dialogue._symbol = symbol;
+            popup_dialogue._map = map;
+            dialogue_background._show();
+        };
+        this.popup_dialogue._hide = function () {
+            this.style.display = 'none';
+            popup_dialogue._title.innerText = '';
+            popup_dialogue._input.value = '';
+            popup_dialogue._input.setCustomValidity('');
+            dialogue_background._hide();
+        };
+        dialogue_background.addEventListener('click', (ev) => {
+            popup_dialogue._hide();
+        });
+    }
+
+    draw() {
+        this.renderer.for_all_elements(0, 0, 0, 0,
+            (type, element, object, intersect) => {
+                if (object instanceof Edge) {
+                    object.draw_memory_volume_overlay(this.renderer,
+                        this.renderer.ctx, this, this.cutoff_high_volume);
+                }
+            }
+        );
+    }
+
+    on_mouse_event(type, ev, mousepos, elements, foreground_elem, ends_drag) {
+        if ((type === 'click' && !ends_drag) || type === 'dblclick') {
+            if (foreground_elem && foreground_elem.data &&
+                foreground_elem.data.attributes &&
+                foreground_elem.data.attributes.volume) {
+                this.parse_volume(foreground_elem.data.attributes.volume, true);
+            }
+        }
+        return false;
+    }
+
+}
+
+class OverlayManager {
+
+    constructor (renderer) {
+        this.renderer = renderer;
+
+        this.memory_volume_overlay = new MemoryVolumeOverlay(this.renderer);
```

Comment:
why construct it here?

Reply:
Would you construct it on first activation? Or externally and fed back in to the manager?

Reply:
externally as adding an overlay to a list of active overlays (or just `this.active_overlay`)

Reply:
The list of active overlays works too, I'll change to that. The thinking was to avoid running the setup code multiple times, but the overhead should be negligible either way.

Reply:
@phschaad are you planning to change this?

Reply:
I am, yes

Reply:
OK, unrequested myself from being a reviewer. Please re-request review when done. :)

Reply:
Done :-)